### PR TITLE
Update java version

### LINKED ISSUE #158: Get required java version from maven project when running github actions
When compiling code and running tests in github actions, java version has to be specified and changed when a new version arrives. Instead of having to change java version each time manually we can configure our actions to read java version from our maven config.
` - name: Get required JDK version from Maven project
      run: |
        JDK_VERSION=$(mvn help:evaluate -Dexpression=maven.compiler.release -q -DforceStdout)
        echo "JDK_VERSION=$JDK_VERSION" >> $GITHUB_ENV
    - name: Set up JDK
      uses: actions/setup-java@v2
      with:
        java-version: ${{ env.JDK_VERSION }}
        distribution: microsoft
        `
 [https://github.com/brunoborges/aks-jvm-benchmark/blob/main/.github/workflows/maven.yml](       https://github.com/brunoborges/aks-jvm-benchmark/blob/main/.github/workflows/maven.yml)

### LINKED ISSUE #197: Update java version
The java version used is not consistent. We would like to have latest java version in both pom.xml, Dockerfile and in github workflow files.
Can have something to do with issue #158 also.

### PR DESCRIPTION
**Pull request by nilssonjn**
Update to java version 21, update dependencies and the dockerfile. Also added script for issue #158

closes #158
closes #197

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -104,7 +103,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-failsafe-plugin</artifactId>
-                <version>3.0.0-M7</version>
+                <version>3.1.0</version>
```
Comment (Ebyrdeu) [inline pom.xml:106]: The latest verison is [3.2.5](https://mvnrepository.com/artifact/org.apache.maven.plugins/maven-failsafe-plugin).
For some reason inteleji sugestion hasn't updated that to 3.2.5 and latest version you can see is 3.1.0 I belive.

File: pom.xml [hunk 2]
```diff
@@ -9,12 +9,11 @@
...
     <properties>
-        <maven.compiler.source>17</maven.compiler.source>
-        <maven.compiler.target>${maven.compiler.source}</maven.compiler.target>
+        <maven.compiler.release>21</maven.compiler.release>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <junit.jupiter.version>5.9.2</junit.jupiter.version>
-        <assertj.core.version>3.23.1</assertj.core.version>
-        <mockito.version>5.2.0</mockito.version>
+        <junit.jupiter.version>5.10.2</junit.jupiter.version>
+        <assertj.core.version>3.25.2</assertj.core.version>
```
Comment (Ebyrdeu) [inline pom.xml:15]: The latest version is [3.25.3](https://mvnrepository.com/artifact/org.assertj/assertj-core)

File: pom.xml [hunk 3]
```diff
@@ -117,7 +116,7 @@
             <plugin>
                 <groupId>org.jacoco</groupId>
                 <artifactId>jacoco-maven-plugin</artifactId>
-                <version>0.8.8</version>
+                <version>0.8.10</version>
```
Comment (Ebyrdeu) [inline pom.xml:119]: The latest version is [0.8.11](https://mvnrepository.com/artifact/org.jacoco/jacoco-maven-plugin)

**Review by Ebyrdeu — CHANGES_REQUESTED** [review]

**Review by Ebyrdeu — APPROVED** [review]

**Review by hannastigland — APPROVED** [review]
Looks correct to me, thank you for contributing!

**Pull request merged**